dashboard/test/models/sections/section_test.rb

```diff
@@ -71,6 +71,78 @@ class SectionTest < ActiveSupport::TestCase
     end
   end
 
+  test 'changing sharing_disabled to true updates user settings' do
+    student = create :student, sharing_disabled: false
+    section = Section.create @default_attrs
```

Comment:
`self.use_transactional_test_case = true` in the [test](https://github.com/code-dot-org/code-dot-org/blob/12d03e49cad4a04de2d226fa5a37bc2e35283d05/dashboard/test/models/sections/section_test.rb#L4) should prevent any state from bleeding across test cases, even where it modifies the rows created in `setup_all`.

I'm not saying you should use `@section` here instead of creating a local one, but it should be fine either way.